### PR TITLE
Add parser umask

### PR DESCRIPTION
**Pull request by wushiqinlou**

### CODE REVIEW

**Comment by wushiqinlou** [comment]
insights-core-assets MR: https://gitlab.cee.redhat.com/insights-release-eng/insights-core-assets/-/merge_requests/230

File: insights/parsers/umask.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+    Sample content::
+
+        0022
+
+    Examples:
+
+        >>> type(umask_obj)
+        <class 'insights.parsers.umask.Umask'>
+        >>> umask_obj['value']
+        '0022'
```
Comment (bfahr) [inline insights/parsers/umask.py:29]: A dictionary really doesn't make sense here, at least the way you have used it.  Why not use `umask -S` and create keys for `user`, `group` and `other`, and then have a tuple or list of the settings for that key?  The way it is currently implemented it simply a copy of the spec.

File: insights/parsers/umask.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+
+        "0022"
+
+    Raises:
+        SkipException: When contents are empty or invalid
+
+    """
+
+    def parse_content(self, content):
+
```
Comment (bfahr) [inline insights/parsers/umask.py:42]: There should be a docstring here to replace the default docstring.  A one-liner is sufficient.

**Review by bfahr — CHANGES_REQUESTED** [review]

File: insights/parsers/umask.py (hunk 1 above)
Comment (wushiqinlou) [reply insights/parsers/umask.py]: @bfahr - Updated

File: insights/specs/default.py [hunk 3]
```diff
@@ -992,6 +992,7 @@ def is_mod_loaded_for_ss(broker):
     tomcat_vdc_fallback = simple_command("/usr/bin/find /usr/share -maxdepth 1 -name 'tomcat*' -exec /bin/grep -R -s 'VirtualDirContext' --include '*.xml' '{}' +")
     tuned_adm = simple_command("/usr/sbin/tuned-adm list")
     udev_fc_wwpn_id_rules = simple_file("/usr/lib/udev/rules.d/59-fc-wwpn-id.rules")
+    umask = simple_command("/usr/bin/umask")
```
[the remark this answers is not in the document]
Comment (wushiqinlou) [reply insights/specs/default.py]: @bfahr - Updated, please have a check.

File: insights/parsers/umask.py [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+from insights.parsers import SkipException
+from insights.specs import Specs
+
+
+@parser(Specs.umask)
+class Umask(Parser, dict):
+    """
+    Parser for ``/usr/bin/umask`` file.
+
+    Sample input is provided in the *Examples*.
```
[the remark this answers is not in the document]
Comment (wushiqinlou) [reply insights/parsers/umask.py]: @bfahr - Updated, please have a check.

File: insights/parsers/umask.py (hunk 2 above)
Comment (wushiqinlou) [reply insights/parsers/umask.py]: @bfahr - Updated, please have a check.

**Review by xiangce — APPROVED** [review]
Looks good to me

**Comment by wushiqinlou** [comment]
since insights client executes commands as root but without the shell, the umask collected by the client is always "u=rwx,g=rwx,o=rwx" (0000).  As insights-client can not collect accurate "umask" when root is logged in, we have to abandon this candidate. Close this MR

**Comment by wushiqinlou** [comment]
Close this PR as the result of "su -l root" is unexpected and dangerous.

**Pull request closed without merging**